detectable/src/main/java/com/synopsys/integration/detectable/detectables/gradle/inspection/parse/GradleReportLineParser.java

```diff
@@ -77,9 +92,53 @@ private List<String> parseGav(String line) {
             }
         }
 
+        String projectName = metadata.getOrDefault(PROJECT_NAME_PREFIX, "orphanProject");
+        String rootProjectName = metadata.getOrDefault(ROOT_PROJECT_NAME_PREFIX, "");
+        String projectParent = metadata.getOrDefault(PROJECT_PARENT_PREFIX, null);
+
+        addRelation(projectParent, projectName, rootProjectName);
+
+        if(gavPieces.size() == 3) {
+            String dependencyGroupName = gavPieces.get(0) + ":" + gavPieces.get(1);
+            if ((cleanedOutput.contains(STRICTLY) || cleanedOutput.contains(REJECT) || cleanedOutput.contains(REQUIRE) || cleanedOutput.contains(PREFER))) {
+                gradleRichVersions.putIfAbsent(dependencyGroupName, gavPieces.get(2));
```

Comment:
Is it correct to do this if the line indeed contains "reject"?

Reply:
Yes it would be correct, gradle dependency resolution is smart enough to reject the version which has been declared and resolve to the next available version. We are saving the version if that dependency has been used by child project so we can update that in such case.